op-e2e/actions/garbage_channel_out.go

```diff
@@ -0,0 +1,223 @@
+package actions
+
+import (
+	"bytes"
+	"compress/gzip"
+	"compress/zlib"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+var ErrNotDepositTx = errors.New("first transaction in block is not a deposit tx")
+var ErrTooManyRLPBytes = errors.New("batch would cause RLP bytes to go over limit")
+
+// WriterApi is the interface shared between `zlib.Writer` and `gzip.Writer`
+type WriterApi interface {
+	Close() error
+	Flush() error
+	Reset(io.Writer)
+	Write([]byte) (int, error)
+}
+
+// GarbageChannelOut is a modified `derive.ChannelOut` that can be configured to behave differently
```

Comment:
What do you think about adding an interface to `op-node` that `derive.ChannelOut` implements and then import that interface here and make a compile time interface assertion that `GarbageChannelOut` implements

Reply:
I like this idea- done 🤝